src/common/snippets/include/snippets/lowered/loop_manager.hpp

```diff
@@ -154,56 +214,139 @@ class LinearIR::LoopManager {
+     * @brief Update Loop ports for several Loops.
+     * @param loop_ids the target Loop IDs
+     * @param actual_port the current port
+     * @param target_ports vector of the new ports (the order is important!)
+     * @param is_entry True if these ports are input, otherwise - output
+     */
     template<typename T>
     void update_loops_port(const std::vector<size_t>& loop_ids, const T& actual_port,
                            const std::vector<T>& target_ports, bool is_entry = true) {
         for (auto loop_id : loop_ids) {
             update_loop_port(loop_id, actual_port, target_ports, is_entry);
         }
     }
-    // Sort Loop Ports by expression locations in Linear IR
+    /**
+     * @brief Sort Loop Ports by expression locations in Linear IR
+     * @param loop_begin_pos the first expression iterator of the Loop
+     * @param loop_end_pos the next iterator after the last expression
+     * @param loop_id target Loop ID
+     */
     void sort_loop_ports(LinearIR::constExprIt& loop_begin_pos, LinearIR::constExprIt& loop_end_pos, size_t loop_id);
-
-    // When the previous expression was replaced with new expressions (decomposition), the method updates the corresponding Loop.
-    // If ports of decomposed expression were the Loop ports, these Loop ports may be updated by parameters `entries` and `exits`
-    // Note: This method should be removed when Softmax decomposition will be moved on data flow pipeline since
-    //       all decompositions should be call on this pipeline
+    /**
+     * @brief When the previous expression was replaced with new expressions (decomposition), the method updates the corresponding Loop.
+     *        If ports of decomposed expression were the Loop ports, these Loop ports may be updated by parameters `entries` and `exits`
+     *        Note: This method should be removed when Softmax decomposition will be moved on data flow pipeline since
+     *              all decompositions should be call on this pipeline
+     * @param new_expr_begin the first expression iterator
+     * @param new_expr_end the next iterator after the last expression
+     * @param decomposed_expr the expression that is decomposed into several other exprs
+     * @param loop_id the new expression will be marked by this loop ID
+     * @param new_entries new input loop ports
+     * @param new_exits new output loop ports
+     */
     void expression_replacement(constExprIt new_expr_begin, constExprIt new_expr_end, const ExpressionPtr& decomposed_expr,
                                 size_t loop_id, const std::vector<ExpressionPort>& new_entries, const std::vector<ExpressionPort>& exits);
+    /**
+     * @brief Find iterators of first entry loop point and last exit point (bounds of Loop) when explicit Loop exprs haven't been inserted in LinearIR
+     *        Note: This method cannot be called when explicit loop exprs are inserted since Loop bounds are defined by LoopBegin and LoopEnd after the insertion!
```

Comment:
When we design a method, we should do our best to guarantee that it yields appropriate and expected result. If we can not guarantee it, we should throw an exception. Unfortunately, it is not enough to write a comment to prevent incorrect usage of a method 🥲 
We need to either generalize it to always work correctly (which is a preferred way) or to throw an exception, when it is used incorrectly

Reply:
LoopBegin/End with a given id can't be inserted at any position, right? (we still nave our entry/exit points). Then do we really need to iterate through the whole IR?
Let's continue this discussion in loop_manager.cpp